BackupActivityTest_Import.testImportReal fails after sharedUserId change

* Caused by #299

```
net.twisterrob.inventory.android.activity.BackupActivityTest_Import > testImportReal[test(AVD) - 5.0.2] FAILED 
	java.lang.SecurityException: Permission Denial: starting Intent { dat=file:///data/data/net.twisterrob.inventory.debug/cache/junit747003320/junit-809493362.tmp typ=application/zip flg=0x10000000 cmp=net.twisterrob.inventory.debug.test/net.twisterrob.inventory.android.test.activity.ScopedStorageSaver (has extras) } from ProcessRecord{3ccb7d85 2900:net.twisterrob.inventory.debug/u0a53} (pid=2900, uid=10053) not exported from uid 10054
```

Also affects:
```
Caused by: java.lang.SecurityException: Permission Denial: starting Intent { flg=0x14000000 cmp=net.twisterrob.inventory.debug.test/net.twisterrob.inventory.android.test.activity.CompatibleLauncher } from ProcessRecord{1e5cd983 4308:net.twisterrob.inventory.debug/u0a199} (pid=4308, uid=10199) not exported from uid 10200
```

--- a/android/src/androidTest/AndroidManifest.xml
+++ b/android/src/androidTest/AndroidManifest.xml
@@ -28,7 +28,10 @@
 		<activity
 			android:name="net.twisterrob.inventory.android.test.activity.CompatibleLauncher"
 			android:label="CompatibleLauncher"
-			/>
+			android:enabled="false"
+			>
+			<!-- This is listed here for documentation, the real registration is in src/debug/AndroidManifest.xml -->
+		</activity>
 
 		<activity
 			android:name="net.twisterrob.inventory.android.test.activity.ScopedStorageSaver"

--- a/android/src/androidTest/AndroidManifest.xml
+++ b/android/src/androidTest/AndroidManifest.xml
@@ -33,7 +33,10 @@
 		<activity
 			android:name="net.twisterrob.inventory.android.test.activity.ScopedStorageSaver"
 			android:label="ScopedStorageSaver"
-			/>
+			android:enabled="false"
+			>
+			<!-- This is listed here for documentation, the real registration is in src/debug/AndroidManifest.xml -->
+		</activity>
 
 	</application>
 

--- a/android/src/androidTest/java/net/twisterrob/inventory/android/test/activity/CompatibleLauncher.java
+++ b/android/src/androidTest/java/net/twisterrob/inventory/android/test/activity/CompatibleLauncher.java
@@ -3,8 +3,6 @@ package net.twisterrob.inventory.android.test.activity;
 import android.app.Activity;
 
 /**
- * Warning: this class might be broken since upgrade to UTP / removal of sharedUserId.
- *
  * Activity that can be used to launch other activities.
  * If this one is launched with {@link android.content.Intent#FLAG_ACTIVITY_NEW_TASK}
  * the other activities started from this are not required to be launched that way, for example:

--- a/android/src/androidTest/java/net/twisterrob/inventory/android/test/activity/ScopedStorageSaver.java
+++ b/android/src/androidTest/java/net/twisterrob/inventory/android/test/activity/ScopedStorageSaver.java
@@ -84,7 +84,7 @@ public class ScopedStorageSaver extends ComponentActivity {
 
 	public static String createZipInDownloads(@NonNull File file)
 			throws PackageManager.NameNotFoundException, UiObjectNotFoundException {
-		Context context = InstrumentationRegistry.getInstrumentation().getContext();
+		Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
 		Uri uri = Uri.fromFile(file);
 		Intent intent = createIntent(context, uri, "application/zip", file.getName())
 				.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

--- a/android/src/androidTest/java/net/twisterrob/inventory/android/upgrade/UpgradeTests.java
+++ b/android/src/androidTest/java/net/twisterrob/inventory/android/upgrade/UpgradeTests.java
@@ -50,8 +50,6 @@ import static net.twisterrob.android.test.matchers.AndroidMatchers.*;
 import static net.twisterrob.test.hamcrest.Matchers.*;
 
 /**
- * Warning: this class might be broken since upgrade to UTP / removal of sharedUserId.
- * <p>
  * <b>These tests need to be excluded from normal {@code gradle connectedCheck} runs.</b>
  * For reason this they require to be run with
  * {@code adb shell am instrument -e class <specific method> -e upgrade true <AndroidJUnitRunner>}.
@@ -89,7 +87,7 @@ public class UpgradeTests {
 
 	@SuppressWarnings("deprecation")
 	@Rule public final androidx.test.rule.ActivityTestRule<CompatibleLauncher> activity =
-			new TestPackageIntentRule<>(CompatibleLauncher.class);
+			new androidx.test.espresso.intent.rule.IntentsTestRule<>(CompatibleLauncher.class);
 
 	@Rule public final IdlingResourceRule drawer = DrawerIdlingResource.rule();
 

--- a/android/src/debug/AndroidManifest.xml
+++ b/android/src/debug/AndroidManifest.xml
@@ -13,6 +13,13 @@
 		tools:replace="android:name"
 		>
 		<!-- lint:AllowBackup REPORT false positive, because main/AndroidManifest.xml is set up. -->
+
+		<!--suppress AndroidDomInspection workaround for https://github.com/TWiStErRob/net.twisterrob.inventory/issues/309 -->
+		<activity
+			android:name="net.twisterrob.inventory.android.test.activity.ScopedStorageSaver"
+			android:label="ScopedStorageSaver"
+			/>
+
 	</application>
 
 </manifest>

--- a/android/src/debug/AndroidManifest.xml
+++ b/android/src/debug/AndroidManifest.xml
@@ -20,6 +20,12 @@
 			android:label="ScopedStorageSaver"
 			/>
 
+		<!--suppress AndroidDomInspection workaround for https://github.com/TWiStErRob/net.twisterrob.inventory/issues/309 -->
+		<activity
+			android:name="net.twisterrob.inventory.android.test.activity.CompatibleLauncher"
+			android:label="CompatibleLauncher"
+			/>
+
 	</application>
 
 </manifest>

